content/en/blog/_posts/2022-03-31-service-ip-dynamic-and-static-allocation.md

```diff
@@ -0,0 +1,97 @@
+an application running on a set of Pods. Services
+can have a `ClusterIP` that is virtual and allows to load-balance traffic across the different Pods.
+
+## How Service ClusterIPs are allocated?
+
+A Service `ClusterIP` can be assigned:
+
+_dynamically_
+: the cluster's control plane automatically picks a free IP address from within the configured IP range for `type: ClusterIP` Services.
+
+_statically_
+: you specify an IP address of your choice, from within the configured IP range for Services.
+
+Across your whole cluster, every Service `ClusterIP` must be unique.
+Trying to create a Service with a `ClusterIP` that has already
+been allocated will return an error.
+
+## Why do you need to reserve Service Cluster IPs?
+
+Because you may want to have Services running in well-known IP addresses, so other components and
+users in the cluster can use them.
+
+The best example is the DNS Service for the cluster. Usually, people assign the 10th IP address in
+the Service IPv4 range to the DNS service. Assuming you configured your cluster with Service IP range
+10.96.0.0/16 and you want your DNS Service IP to be 10.96.0.10, you'd have to create a Service like
+this:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: CoreDNS
+  name: kube-dns
+  namespace: kube-system
+spec:
+  clusterIP: 10.96.0.10
+  ports:
+  - name: dns
+    port: 53
+    protocol: UDP
+    targetPort: 53
+  - name: dns-tcp
+    port: 53
+    protocol: TCP
+    targetPort: 53
+  selector:
+    k8s-app: kube-dns
+  type: ClusterIP
+```
+
+but as I explained before, the IP address 10.96.0.10 has not been reserved; if other Services are created
+before or in parallel with dynamic allocation, there is a chance they can allocate this IP, hence,
+you will not be able to create the DNS Service because it will fail with a conflict error.
+
+## How can you avoid Service ClusterIP conflicts? {#avoid-ClusterIP-conflict}
+
+Since Kubernetes 1.24, a new Feature Gate `ServiceIPStaticSubrange` allows to use a different IP
+allocation strategy for Services, reducing the risk of collision.
+
+The `ClusterIP` range will be divided, based on the formula `min(max(16, cidrSize / 16), 256)`,
+described as _never less than 16 or more than 256 with a graduated step between them_.
+
+Dynamically IP assignment will use the upper band by default, once this has been exhausted it will
+use the lower range. This will allow users to use static allocations on the lower band with a low
+risk of collision.
+
+Example:
+
+- Service IP CIDR block: 192.168.0.0/16
+- Range Size: 2<sup>16</sup> - 2 = 65534
+- Band Offset: `min(max(16,65536/16),256)` = `min(4096,256)` = 256
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.1.0
+
+{{< figure src="static/images/blog/2022-03-31-service-ip-dynamic-and-static-allocation/serviceip-allocator.svg" alt="Kubernetes Service ClusterIP allocator" >}}
```

Comment:
not working great :/ 

https://deploy-preview-32606--kubernetes-io-main-staging.netlify.app/blog/2022/03/31/service-ip-dynamic-and-static-allocation/

Reply:
Try this:
```suggestion
{{< figure src="/images/blog/2022-03-31-service-ip-dynamic-and-static-allocation/serviceip-allocator.svg" alt="Service ClusterIP allocator" >}}
```